catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/IdentificationPlugin.java

```diff
@@ -269,39 +275,36 @@ private String getRegistryId(Metacard mcard) {
                 .toString();
     }
 
-    public void init()
-            throws UnsupportedQueryException, SourceUnavailableException, FederationException,
-            PluginExecutionException {
+    public void init() {
+        try {
+            List<Metacard> registryMetacards;
+            Filter registryFilter = filterBuilder.attribute(Metacard.TAGS).is().like()
+                    .text(RegistryConstants.REGISTRY_TAG);
+            QueryImpl query = new QueryImpl(registryFilter);
+            query.setPageSize(1000);
+            QueryRequest request = new QueryRequestImpl(query);
 
-        List<Metacard> registryMetacards;
-        Filter registryFilter = filterBuilder.attribute(Metacard.TAGS)
-                .is()
-                .like()
-                .text(RegistryConstants.REGISTRY_TAG);
-        QueryImpl query = new QueryImpl(registryFilter);
-        query.setPageSize(1000);
-        QueryRequest request = new QueryRequestImpl(query);
+            request.getProperties().put(SecurityConstants.SECURITY_SUBJECT,
+                    Security.runAsAdmin(() -> Security.getInstance().getSystemSubject()));
 
-        request.getProperties()
-                .put(SecurityConstants.SECURITY_SUBJECT,
-                        Security.runAsAdmin(() -> Security.getInstance()
-                                .getSystemSubject()));
+            QueryResponse response = catalogFramework.query(request);
 
-        QueryResponse response = catalogFramework.query(request);
+            if (response == null) {
+                throw new PluginExecutionException(
+                        "Plugin failed to initialize plugin, unable to query identity node.");
+            }
 
-        if (response == null) {
-            throw new PluginExecutionException(
-                    "Plugin failed to initialize plugin, unable to query identity node.");
+            registryMetacards = response.getResults().stream().map(Result::getMetacard)
+                    .collect(Collectors.toList());
+            registryIds.addAll(registryMetacards.stream().map(this::getRegistryId)
+                    .collect(Collectors.toList()));
+        } catch (UnsupportedQueryException | SourceUnavailableException | FederationException | PluginExecutionException e) {
+            executorService.schedule(this::init, RETRY_INTERVAL, TimeUnit.SECONDS);
```

Comment:
Same comment as earlier where I think you need some sort of "try N times" semantics.